### DISCUSSION TITLE
Run an operation against a single agent, when multiple connected?

### DISCUSSION THREAD

**Discussion by koncentrat11** [mitre/caldera#2899] [General]
If I have multiple agents and want to run an operation against one agent, how do I accomplish that? I can't seem to find a way. I even tried to create a new operation and select "No adversary" to just run commands manually, but no matter what agent I select in the dropdown configuring the command, it just defaults to the first agent.

**Comment by nipunnegi2** [comment]
Is this issue resolved?

**Comment by keysersoze01201** [comment]
I have this exact same issue.  I want to start running certain abilities against only a single agent at a certain stage of the operation.   As I am running a lot of abilities and once i laterally move i don't want the next set of abilities to run against all the systems.  I also dont want to create another adversary and run that separately.  

I have even gone into the level of editing the yml files to include as section under requirements like:
requirements:
  - module: fact
    trait: host.hostname
    value: hostnameofsystem

But this doesn't work.

**Comment by keysersoze01201** [comment]
So this is not possible I assume?

**Reply by nipunnegi2** [reply]
Yes , I think this is not possible. So I am working to make it automated end to end. But I think this thing is not possible.

**Reply by keysersoze01201** [reply]
@nipunnegi2 Thank you for the response.  It would be a nice feature if we could edit the .yml file and add like host.hostname = hostname to run certain abilities against certain hosts only.
